src/app/public/modules/datepicker/datepicker.component.ts

```diff
@@ -1,67 +1,83 @@
 import {
+  ChangeDetectionStrategy,
   Component,
   EventEmitter,
-  ViewChild
+  OnDestroy,
+  ViewChild,
+  ChangeDetectorRef
 } from '@angular/core';
 
-import { Subject } from 'rxjs/Subject';
-
-import { SkyDatepickerCalendarComponent } from './datepicker-calendar.component';
-
 import {
-  SkyDropdownComponent,
   SkyDropdownMessage,
   SkyDropdownMessageType
 } from '@skyux/popovers';
 
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subject
+} from 'rxjs/Subject';
+
+import {
+  SkyDatepickerCalendarComponent
+} from './datepicker-calendar.component';
+
 @Component({
   selector: 'sky-datepicker',
   templateUrl: './datepicker.component.html',
-  styleUrls: ['./datepicker.component.scss']
+  styleUrls: ['./datepicker.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class SkyDatepickerComponent {
-  @ViewChild(SkyDatepickerCalendarComponent)
-  public calendar: SkyDatepickerCalendarComponent;
+export class SkyDatepickerComponent implements OnDestroy {
+  public get disabled(): boolean {
+    return this._disabled;
+  }
+
+  public set disabled(value: boolean) {
+    this._disabled = value;
+    this.changeDetector.markForCheck();
+  }
+
+  public get dropdownController(): Observable<SkyDropdownMessage> {
+    return this._dropdownController;
+  }
 
-  @ViewChild(SkyDropdownComponent)
-  public dropdown: SkyDropdownComponent;
+  public set selectedDate(value: Date) {
+    this.calendar.writeValue(value);
+  }
 
-  public dropdownController = new Subject<SkyDropdownMessage>();
-  public dateChanged: EventEmitter<Date> = new EventEmitter<Date>();
-  public disabled: boolean;
-  private _startingDay: number = 0;
+  public dateChange = new EventEmitter<Date>();
   public maxDate: Date;
   public minDate: Date;
+  public startingDay: number;
 
-  public get startingDay(): number {
-    return this._startingDay;
-  }
-  public set startingDay(value: number) {
-    this._startingDay = value;
-  }
+  @ViewChild(SkyDatepickerCalendarComponent)
+  private calendar: SkyDatepickerCalendarComponent;
 
-  public dateSelected(newDate: Date) {
-    this.dateChanged.emit(newDate);
-    this.dropdownController.next({
-      type: SkyDropdownMessageType.Close
-    });
-  }
+  private _disabled = false;
+  private _dropdownController = new Subject<SkyDropdownMessage>();
 
-  public setSelectedDate(newDate: Date) {
```

Comment:
Removing a public property. Is this deliberate?

Reply:
I switched the public method to a setter field (`public set selectedDate`). It was only being used by the `datepicker-input` directive.